src/macros.rs

```diff
@@ -0,0 +1,9 @@
+
+#[macro_export]
+/// Constructs a `&'static CStr` literal.
+macro_rules! cstr {
+    ($s: tt) => {
+        // TODO: verify that $s is a string literal without nuls
+        unsafe { ::std::ffi::CStr::from_ptr(concat!($s, "\0").as_ptr() as *const _) }
```

Comment:
Sidenote: I'd prefer `CString::new("...").unwrap().as_ptr()` to that macro.

Reply:
wouldn't `CString::new("...").unwrap().as_ptr()` deallocate at the end of the block due to itself being dropped too early?

Reply:
Oh your're right, I meant `CString::new("...").unwrap().into_raw()` ("Consumes the CString and transfers ownership of the string to a C caller.").

Reply:
I've removed that macro in 09883574, which results in a minimal merge conflict.

Reply:
Nice! Creating `CStr` directly is much nicer, I'll merge those changes :)

Reply:
by the way, it seems adding a nul terminator is needed for those strings (otherwise it will error)